lib/local-cluster/lib/Cardano/Wallet/Launch/Cluster/Monitoring/Http/Client.hs

```diff
@@ -0,0 +1,172 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE NumericUnderscores #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+module Cardano.Wallet.Launch.Cluster.Monitoring.Http.Client
+    ( withHttpClient
+    , RunQuery (..)
+    , Query (..)
+    , MsgClient (..)
+    )
+where
+
+import Prelude
+
+import Cardano.Wallet.Launch.Cluster.Monitoring.Http.API
+    ( MonitorApi
+    )
+import Cardano.Wallet.Launch.Cluster.Monitoring.Http.SendFaucetAssets
+    ( SendFaucetAssets (..)
+    , WithNetwork (..)
+    )
+import Cardano.Wallet.Primitive.NetworkId
+    ( HasSNetworkId
+    )
+import Cardano.Wallet.Primitive.Types.Address
+    ( Address
+    )
+import Cardano.Wallet.Primitive.Types.TokenBundle
+    ( TokenBundle
+    )
+import Control.Monad
+    ( unless
+    , void
+    )
+import Control.Monad.Cont
+    ( ContT (..)
+    )
+import Control.Monad.IO.Class
+    ( liftIO
+    )
+import Control.Retry
+    ( RetryPolicyM
+    , RetryStatus (..)
+    , capDelay
+    , exponentialBackoff
+    , recoverAll
+    )
+import Control.Tracer
+    ( Tracer
+    , traceWith
+    )
+import Network.HTTP.Client
+    ( ManagerSettings (..)
+    , defaultManagerSettings
+    , newManager
+    , responseTimeoutNone
+    )
+import Servant
+    ( NoContent
+    , (:<|>) (..)
+    )
+import Servant.Client
+    ( BaseUrl (..)
+    , ClientM
+    , Scheme (..)
+    , client
+    , mkClientEnv
+    , runClientM
+    )
+import UnliftIO
+    ( MonadUnliftIO
+    , UnliftIO (..)
+    , askUnliftIO
+    , throwIO
+    )
+
+-- | Queries that can be sent to the monitoring server via HTTP.
+data Query a where
+    ReadyQ :: Query Bool
+    SendFaucetAssetsQ
+        :: Int
+        -> [(Address, (TokenBundle, [(String, String)]))]
+        -> Query ()
+
+data Client n = Client
+    { ready :: ClientM Bool
+    , sendFaucetAssets :: WithNetwork SendFaucetAssets n -> ClientM NoContent
+    }
+
+mkClient :: HasSNetworkId n => MonitorApi n -> Client n
+mkClient p =
+    let ready :<|> sendFaucetAssets = client p
+    in  Client
+            { ready
+            , sendFaucetAssets
+            }
+
+data AnyQuery = forall a. Show a => AnyQuery (Query a)
+
+instance Show AnyQuery where
+    show (AnyQuery ReadyQ) = "Ready"
+    show (AnyQuery (SendFaucetAssetsQ n xs)) =
+        "SendFaucetAssets " <> show n <> " " <> show xs
+
+-- | Run any query against the monitoring server.
+newtype RunQuery m = RunQuery (forall a. Show a => Query a -> m a)
+
+data MsgClient
+    = MsgClientStart
+    | MsgClientReq AnyQuery
+    | MsgClientRetry AnyQuery
+    | MsgClientDone
+    deriving stock (Show)
+
+-- | Produce a closure over the http client of an http monitoring server that
+-- can be used to query the server.
+withHttpClient
+    :: (MonadUnliftIO m, HasSNetworkId n)
+    => MonitorApi n
+    -> Tracer m MsgClient
+    -> Int
```

Comment:
```suggestion
    -> PortNumber
```

I think that it's a good idea to use the standard type from `Network.Socket` here.